srcalc/src/main/java/gov/va/med/srcalc/web/view/admin/EditMultiSelectVariable.java

```diff
@@ -0,0 +1,168 @@
+     */
+    @Override
+    public String getNewViewName()
+    {
+        return Views.NEW_MULTI_SELECT_VARIABLE;
+    }
+    
+    @Override
+    public ImmutableSortedSet<RetrievalEnum> getAllRetrievers()
+    {
+        return ImmutableSortedSet.of(RetrievalEnum.GENDER);
+    }
+    
+    /**
+     * Returns all possible display types for user selection.
+     * @return an immutable collection
+     */
+    public final ImmutableCollection<MultiSelectVariable.DisplayType> getAllDisplayTypes()
+    {
+        return ImmutableList.copyOf(MultiSelectVariable.DisplayType.values());
+    }
+    
+    /**
+     * Returns the DisplayType to set on the variable.
+     * @see MultiSelectVariable#getDisplayType()
+     */
+    public final MultiSelectVariable.DisplayType getDisplayType()
+    {
+        return fDisplayType;
+    }
+    
+    /**
+     * Sets the DisplayType to set on the variable.
+     * @see MultiSelectVariable#getDisplayType()
+     */
+    public final void setDisplayType(final MultiSelectVariable.DisplayType displayType)
+    {
+        fDisplayType = displayType;
+    }
+    
+    /**
+     * Returns the options to set on the variable. Note that {@link
+     * MultiSelectVariable}s actually have collections of {@link MultiSelectOption},
+     * but since those objects effectively only contain Strings, we represent
+     * them as Strings here for easy use in views.
+     * @see MultiSelectVariable#getOptions()
+     * @return a mutable List
+     */
+    public final List<String> getOptions()
+    {
+        return fOptions;
+    }
+    
+    /**
+     * Like {@link #getOptions()} but with trailing blanks trimmed off. (The
+     * HTTP form submission may create spurious trailing blank options.) The
+     * returned list may still contain empty strings in the middle.
+     * @return an ImmutableList
+     */
+    public final ImmutableList<String> getTrimmedOptions()
+    {
+        // Start with a copy of all the options.
+        final ArrayList<String> trimmed = new ArrayList<>(fOptions);
+        
+        // Iterate from the end, removing as we go.
+        for (int i = fOptions.size() - 1; i >= 0; --i)
+        {
+            if (Strings.isNullOrEmpty(trimmed.get(i)))
+            {
+                trimmed.remove(i);
+            }
+            // Break at the first non-empty String.
+            else
+            {
+                break;
+            }
+        }
+        
+        return ImmutableList.copyOf(trimmed);
+    }
+    
+    /**
+     * Returns the options as a List of {@link MultiSelectOption}, the type
+     * needed to actually create the variable. Internally uses {@link
+     * #getTrimmedOptions()} to omit any trailing blanks.
+     * @return an ImmutableList
+     */
+    protected ImmutableList<MultiSelectOption> getMultiSelectOptions()
+    {
+        final ImmutableList<String> trimmed = getTrimmedOptions();
+        final ArrayList<MultiSelectOption> options = new ArrayList<>(trimmed.size());
```

Comment:
Just for a low-cost performance gain. We know exactly what capacity will be needed.